src/main/java/com/yugabyte/sample/apps/SqlEventCounter.java

```diff
@@ -0,0 +1,196 @@
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied.  See the License for the specific language governing permissions and limitations
+// under the License.
+//
+
+package com.yugabyte.sample.apps;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.concurrent.atomic.AtomicIntegerArray;
+import java.util.concurrent.atomic.AtomicLong;
+
+import java.util.concurrent.ThreadLocalRandom;
+
+import org.apache.log4j.Logger;
+
+/*
+ * Sometimes, applications want to track the number of times a particular event
+ * has occurred. Examples include user actions like clicks, purchases or
+ * page views.
+ *
+ * This app helps understand whether the new clockbound clock
+ * helps improve the performance of this workload.
+ *
+ * Database Configuration:
+ *  configure with wallclock and compare the metrics with
+ *  a clockbound clock configuration.
+ *  Not much variance is expected in the metrics.
+ *
+ * Setup:
+ * 1. Create a counters TABLE with columns (event INT, counter INT).
+ * 2. Insert 1000 counters with event 0 to 999 initialized to zero.
+ *
+ * Worklaod:
+ * We only run write threads, no read threads.
+ * Each write thread,
+ * 1. Starts a repeatable read transaction.
+ * 2. Reads the counter of a random event.
+ * 3. Verifies that the counter is not stale.
+ * 4. Increments the counter for the picked event.
+ * 5. Commits the transaction.
+ * 6. Updates the latest counter value in the cache.
+ */
+public class SqlEventCounter extends AppBase {
+  private static final Logger LOG = Logger.getLogger(SqlEventCounter.class);
+
+  // Static initialization of this app's config.
+  static {
+    // Only use 10 writer threads to avoid overloading the system.
+    // In real life, there are many more threads but there are other
+    // things to do too.
+    appConfig.readIOPSPercentage = -1;
+    appConfig.numReaderThreads = 0;
+    appConfig.numWriterThreads = 10;
+    // Disable number of keys.
+    appConfig.numKeysToRead = -1;
+    appConfig.numKeysToWrite = -1;
+    // Run the app for 1 minute.
+    appConfig.runTimeSeconds = 60;
+    // Report restart read requests metric by default.
+    appConfig.restartReadsReported = true;
+    // Avoid load balancing errors.
+    appConfig.loadBalance = false;
+    appConfig.disableYBLoadBalancingPolicy = true;
+  }
+
+  // The default table name to create and use for ops.
+  private static final String DEFAULT_TABLE_NAME = "event_counters";
+
+  // The number of unique events to track.
+  private static final int NUM_EVENTS = 1000;
+
+  // Contains the latest updated counter indexed by event.
+  private AtomicIntegerArray counters = new AtomicIntegerArray(NUM_EVENTS);
+
+  // Shared counter to store the number of stale reads.
+  private static AtomicLong numStaleReads = new AtomicLong(0);
```

Comment:
all atomic counters here might be also final, small change , but good code style